### Data analysis tutorials

If you haven't done much exploratory data analysis, here are some good guides to 
learn the basics of data loading, manipulating/summarizing data tables, and making
plots. If you have intermediate experience but would like to fill in a few gaps, you
can jump to the sections in these guides that are most applicable to you.

## Guides for R

https://r4ds.had.co.nz/explore-intro.html \
In general, using the "tidyverse" libraries is a highly-recommended way to process and
visualize data in an organized and easy-to-learn way. For complete beginners with little
previous coding background, we also generally advise R as being quicker to learn for data 
analysis, whereas Python and its associated packages like Pandas can be very powerful but
have a less intuitive syntax.

Many specialized packages exist for analyzing gene expression matrices specifically such as DEseq2 
(https://bioinformatics-core-shared-training.github.io/cruk-summer-school-2018/RNASeq2018/html/04_DE_analysis_with_DESeq2.nb.html)
However, for those with less coding experience an alternative is to use the basic plotting commands
applicable to any kind of data table but focusing on particular genes, groups of genes, or experimental
conditions (e.g. https://liz-is.github.io/R-genomics/06-rnaseq-data.html)


## Guides for Python

https://datacarpentry.org/python-ecology-lesson/ \
The Pandas package in Python greatly faciltiates doing analyses on datasets formated as tables.
The two most popular packages in Python for making plots are matplotlib, which borrows its interface
from Matlab, and plotnine, which borrows its interface from R. Because Matlab structures can be
complex to learn, we recommended plotnine for beginners.


## Guides for other languages

https://orange.biolab.si/screenshots/ \  
Orange provides a graphical interface method of analyzing data, with no explicit coding involved
at all. 
	
